src/MonthList/index.js

```diff
@@ -32,35 +35,41 @@ export default class MonthList extends Component {
     overscanMonthCount: PropTypes.number,
     rowHeight: PropTypes.number,
     selectedDate: PropTypes.instanceOf(Date),
+    showMonthLabels: PropTypes.bool,
     showOverlay: PropTypes.bool,
     theme: PropTypes.object,
     today: PropTypes.instanceOf(Date),
     width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
   };
-  state = {
-    scrollTop: this.getDateOffset(this.props.scrollDate),
-  };
   cache = {};
   memoize = function(param) {
     if (!this.cache[param]) {
-      const {locale: {weekStartsOn}} = this.props;
+      const {locale: {weekStartsOn}, showMonthLabels} = this.props;
       const [year, month] = param.split(':');
-      const result = getMonth(year, month, weekStartsOn);
-      this.cache[param] = result;
+      this.cache[param] = showMonthLabels ?
+        getMonthWithPadding(year, month, weekStartsOn) : getMonth(year, month, weekStartsOn);
     }
     return this.cache[param];
   };
   monthHeights = [];
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      scrollTop: this.getDateOffset(this.props.scrollDate),
+    };
+  }
+
   _getRef = (instance) => { this.VirtualList = instance; }
 
   getMonthHeight = (index) => {
     if (!this.monthHeights[index]) {
-      let {locale: {weekStartsOn}, months, rowHeight} = this.props;
+      let {locale: {weekStartsOn}, months, rowHeight, showMonthLabels} = this.props;
       let {month, year} = months[index];
-      let weeks = getWeeksInMonth(month, year, weekStartsOn, index === months.length - 1);
-      let height = weeks * rowHeight;
-      this.monthHeights[index] = height;
+      let weeks = showMonthLabels ?
```

Comment:
Given all the ternaries I'm seeing, wonder if it'd make sense to extract the varying behavior behind an interface.

Reply:
I was thinking about adapting this functionality into a new recompose but it seemed like a very large undertaking. You aren't wrong though.

Reply:
See `withRange` as an example.